test/test_kubeclient.rb

```diff
@@ -406,6 +406,15 @@ def test_api_bearer_token_file_success
     assert_equal(1, pods.size)
   end
 
+  def test_proxy_url
+    client = Kubeclient::Client.new 'http://host:8080', 'v1'
+    assert_equal('http://host:8080/api/v1/proxy/namespaces/ns/services/srvname:5001-tcp',
+                 client.proxy_url('services', 'srvname', '5001-tcp', 'ns'))
+
+    assert_equal('http://host:8080/api/v1/proxy/pods/srvname:5001-tcp',
```

Comment:
@jimmidyson how do you specify what port to use with `/api/v1/namespaces/{namespace}/pods/{name}/proxy/{path:*}` ?